### PR TITLE
Primary commit towards improved team communications. Add new events a…

### PR DESCRIPTION
**Pull request by tykling**
…pp to handle team notifications when various events happen, with a Type model which contain event types and a Routing model which controls routing of events to teams. Add shortslug for Camp and Team models. events.handler.py contains the code for sending irc and email notifications for teams. The first two eventtypes have been added in datamigrations, 'ticket_created' and 'public_credit_name_changed', and the tickets and profile apps have been adjusted accordingly. Team IRC channels can be marked as managed and if so the IRC bot will register the team channel with ChanServ if possible. Team IRC channels can be marked as private and the bot will set invite only and maintain an ACL with team members. Users can set their NickServ username in their profile to get on the ACL. Rework all team views and templates. Remove TeamArea model and make Team have an FK to Camp directly. Add docstrings a whole bunch of places. Move signal handlers to apps.py and signal_handlers.py in a few apps. Add basic team mailing list handling, more work to be done. Update bootstrap-devsite script to add more teammembers and add some team event routing for the two eventtypes we have.

### CODE REVIEW

**Comment by tykling** [comment]
this is not ready to be deployed to master, we use this PR to track progress and issues

File: src/events/models.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from utils.models import CreatedUpdatedModel
+from teams.models import Team
+
+class Type(CreatedUpdatedModel):
+    """
+    The events.Type model contains different types of system events which can happen.
+    New event types should be added in data migrations.
+    The following types are currently used in the codebase:
+    - ticket_created: Whenever a new ShopTicket is created
+    - public_credit_name_changed: Whenever a user changes public_credit_name in the profile
```
Comment (valberg) [inline src/events/models.py]: What is the idea of having this in the database if we want to enforce that it requires a data migration to create an event?

Comment (tykling) [reply src/events/models.py]: they have properties so it would get messy to use a choicefield

Comment (valberg) [reply src/events/models.py]: After a discussion on IRC we decided to keep it like this :)

**Pull request merged**